axelor-account/src/main/java/com/axelor/apps/account/service/TaxPaymentMoveLineServiceImpl.java

```diff
@@ -48,10 +57,61 @@ public TaxPaymentMoveLine getReverseTaxPaymentMoveLine(TaxPaymentMoveLine taxPay
             taxPaymentMoveLine.getTaxRate(),
             taxPaymentMoveLine.getDetailPaymentAmount().negate(),
             taxPaymentMoveLine.getDate());
-    reversetaxPaymentMoveLine = this.computeTaxAmount(reversetaxPaymentMoveLine);
+    reversetaxPaymentMoveLine.setTaxAmount(taxPaymentMoveLine.getTaxAmount().negate());
     reversetaxPaymentMoveLine.setIsAlreadyReverse(true);
     reversetaxPaymentMoveLine.setVatSystemSelect(taxPaymentMoveLine.getVatSystemSelect());
     taxPaymentMoveLine.setIsAlreadyReverse(true);
     return reversetaxPaymentMoveLine;
   }
+
+  @Override
+  public TaxPaymentMoveLine createTaxPaymentMoveLineWithFixedAmount(
+      Invoice invoice,
+      BigDecimal paymentRatio,
+      int vatSystemSelect,
+      MoveLine invoiceMoveLine,
```

Comment:
`invoiceMoveLine`, `taxLine`, `reconcile`, `paymentRatio` can be null and will trigger a NPE if it's the cases, please manage them.